Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -677,6 +677,13 @@ def decompileComponents(self, data, glyfTable):
 			haveInstructions = haveInstructions | haveInstr
 			self.components.append(component)
 		if haveInstructions:
+			if not data:
+				log.warning(
+					"Composite glyph flags announce presence of instructions, but there are none. %s",
+					[c.glyphName for c in self.components]
+				)
+				return
```

Comment:
We usually don't do these kinds of optimizations or fixes upon decompile. Only on compile. Maybe you should warn but keep the font as is